static/js/subs.js

```diff
@@ -785,6 +785,17 @@ $(function () {
         var search_term = user_list.expectOne().val().trim();
         var search_terms = search_term.toLowerCase().split(",");
 
+        // create a map of labels
+        var users_map = $('#user-checkboxes')
```

Comment:
Why would you have to use jQuery for that if you're re-rendering the template?

The slowness for large teams is probably due to the (un-batched) DOM manipulation. If you do all the DOM manipulation in one go it'll likely speed things up.